### PR TITLE
[Enablement Experience] Avoiding failure edge case

### PR DESCRIPTION
**Pull request by Paramadon**
*Issue #, if available:*
There is a small fail case that can happen during this test which is unlikely maybe 10% of the time. The test would fail because the operator would change the deployment while midst updating deployment.
*Description of changes:*
This change just adds a wait condition for the deployment to be fully ready

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: integration-tests/manifests/annotations/validate_annotations_namespace_test.go [hunk 1]
```diff
@@ -186,22 +186,38 @@ func TestAutoAnnotationForManualAnnotationRemoval(t *testing.T) {
...
+
+		if deployment.Status.AvailableReplicas == *deployment.Spec.Replicas && deployment.Status.UpdatedReplicas == *deployment.Spec.Replicas {
+			if deployment.Status.Replicas == deployment.Status.AvailableReplicas {
+				fmt.Println("All pods are fully ready and no pods are terminating.")
+				break
+			}
+		}
+
+		// Sleep for a short interval before checking again
+		time.Sleep(5 * time.Second)
```
Comment (sethAmazon) [inline integration-tests/manifests/annotations/validate_annotations_namespace_test.go:215]: Should we be doing wait groups instead of sleep?

Comment (Paramadon) [reply integration-tests/manifests/annotations/validate_annotations_namespace_test.go:215]: Hmm I think wait groups might not be the best approach for this scenario because we're dealing with a loop that waits for a certain condition to be met, rather than coordinating multiple concurrent tasks.

Comment (Paramadon) [reply integration-tests/manifests/annotations/validate_annotations_namespace_test.go:215]: Wait groups might not be the most suitable approach for this scenario because we're dealing with a loop that waits for a certain condition to be met, rather than coordinating multiple concurrent tasks. It is a single goroutine that runs in a loop.
